### PR TITLE
Improving HTML5 validation and adding php validation

### PR DESCRIPTION
**Pull request by tmadhuri**

### CODE REVIEW

File: app/views/missing/report.php [hunk 1]
```diff
@@ -136,7 +138,7 @@
                                         <label for="lname">Last name </label>
                                         <input type="text" class="form-control" placeholder="Enter Last Name" id="lname" name="person[lname]" required /><br>
                                         <label for="dob">Date of birth </label>
-                                        <input type="date" placeholder="Enter your DOB" class="form-control" id="dob" type="date" name="person[dob]" /><br>
+                                        <input type="text" placeholder="Enter your DOB (yyyy-mm-dd)" class="form-control" id="dob" name="person[dob]" pattern="(0?[1-9]|[12][0-9]|3[01])[\/\-](0?[1-9]|1[012])[\/\-]\d{4}" /><br>
```
Comment (gvishal) [inline app/views/missing/report.php]: I am unable to add a new missing person report. Below error keeps propping up.
Removing the pattern attribute works.

screenshot from 2015-04-05 03 18 20

Comment (nisargjhaveri) [inline app/views/missing/report.php]: The pattern matches `dd-mm-yyyy`. It should be changed to match `yyyy-mm-dd`.

File: app/controllers/missing.php [hunk 2]
```diff
@@ -10,16 +10,87 @@ function index() {
...
         $reported = null;
 
         if ( isset($_POST['report']) ) {
-            $reported = $this->missing_model->report(
-                $_POST['person'],
-                $_POST['person_contact'],
-                $_POST['person_detail'],
-                $_POST['contact_whom']
-            );
+            if($this->valid($_POST['report'])) {
```
Comment (gvishal) [inline app/controllers/missing.php]: valid() method takes no arguments.

File: app/controllers/missing.php [hunk 3]
```diff
@@ -10,16 +10,87 @@ function index() {
...
+        } 
+        else {
+            return false;
+        }
+
+        return true;
+    }
+
+    function valid() {
+        if( !empty($_POST['person']['dob'])) {
```
Comment (gvishal) [inline app/controllers/missing.php:38]: Dob validation at backend doesn't seem to work, was able to add a record without any dob filled and pattern attribute removed. Record got inserted with DOB 0000-00-00.

Comment (nisargjhaveri) [inline app/controllers/missing.php:38]: This is probably because the next line (line 39). We are not checking the return value of `validateDate`!

**Comment by nisargjhaveri** [comment]
@tmadhuri, from next time please comment after you update PR, as we don't get notified when more commits are pushed to an existing PR.

File: app/views/missing/search.php [hunk 4]
```diff
@@ -145,7 +147,7 @@
                                     </div>
                                     <div class="panel-body" id="panelBody2">
                                         <label for="phone_no">Phone number </label>
-                                        <input placeholder="Enter the Phone Number" class="form-control" id="phone_no" name="person_contact[phone_no]" /><br>
+                                        <input placeholder="Enter the Phone Number" class="form-control" id="phone_no" name="person_contact[phone_no]" pattern="\d{10}" /><br>
```
Comment (nisargjhaveri) [inline app/views/missing/search.php]: @tmadhuri, you forgot to update this.

File: app/views/missing/search.php [hunk 5]
```diff
@@ -94,7 +96,7 @@
             </li>
 
             <li>
-                <a href="<?php echo base_url(); ?>requests/">
+                <a href="#">
```
Comment (nisargjhaveri) [inline app/views/missing/search.php:99]: No need of this change.

File: app/views/missing/report.php [hunk 6]
```diff
@@ -94,7 +96,7 @@
             </li>
 
             <li>
-                <a href="<?php echo base_url(); ?>requests/">
+                <a href="#">
```
Comment (nisargjhaveri) [inline app/views/missing/report.php:99]: No need of this change.

**Comment by nisargjhaveri** [comment]
Fixed and merged with f0ada4dd5421bbc31029594c2f838bc5cc0c50b5.

**Pull request closed without merging**